### PR TITLE
Adding 2.2.0 Bwc version to main

### PR DESCRIPTION
**Pull request by saratvemulapalli**
Signed-off-by: Sarat Vemulapalli <vemulapallisarat@gmail.com>

### Description
Prepping for 2.1.0 release, Adding 2.2.0 Bwc version to main.
 
### Issues Resolved
Part of https://github.com/opensearch-project/OpenSearch/issues/3636
 
### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by saratvemulapalli** [comment]
> ❌   Gradle Check failure [cdacd91](https://github.com/opensearch-project/OpenSearch/commit/cdacd910015804d32ef8e3bd69d9b9d04701967c) [Log 6265](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_6265.log)
> 
> [Reports 6265](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_6265_reports.zip)

Will have to wait for this PR to be merged: https://github.com/opensearch-project/OpenSearch/pull/3672

File: server/src/main/java/org/opensearch/Version.java [hunk 1]
```diff
@@ -92,6 +92,7 @@ public class Version implements Comparable<Version>, ToXContentFragment {
     public static final Version V_2_0_1 = new Version(2000199, org.apache.lucene.util.Version.LUCENE_9_1_0);
     public static final Version V_2_0_2 = new Version(2000299, org.apache.lucene.util.Version.LUCENE_9_1_0);
     public static final Version V_2_1_0 = new Version(2010099, org.apache.lucene.util.Version.LUCENE_9_3_0);
```
Comment (mch2) [inline server/src/main/java/org/opensearch/Version.java:94]: Not sure if this has any impact, but V_2_1_0 will be using LUCENE_9_2_0 with - https://github.com/opensearch-project/OpenSearch/pull/3676

Comment (saratvemulapalli) [reply server/src/main/java/org/opensearch/Version.java]: It should. I've fixed it.

**Comment by saratvemulapalli** [comment]
start gradle check

**Review by tlfeng — APPROVED** [review]

**Pull request merged**